db/dbsession.go

```diff
@@ -12,14 +12,28 @@ var DynamoDB *dynamodb.DynamoDB
 
 // GetDynamoSession returns a new dynamodb session
 func GetDynamoSession(accessKeyID, secretAccessKey, region string) *dynamodb.DynamoDB {
-	token := ""
-	creds := credentials.NewStaticCredentials(accessKeyID, secretAccessKey, token)
-	_, err := creds.Get()
-	if err != nil {
-		panic(err)
+	session_config := session.Options{}
+	if accessKeyID != "" || secretAccessKey != "" {
+		token := ""
+		creds := credentials.NewStaticCredentials(accessKeyID, secretAccessKey, token)
+		_, err := creds.Get()
+		if err != nil {
+			panic(err)
+		}
+		session_config.Config.Credentials = creds
 	}
 
-	DynamoDB = dynamodb.New(session.New(&aws.Config{Region: &region}))
+	if region != "" {
+		session_config.Config.Region = &region
+	} else {
+		session_config.SharedConfigState = session.SharedConfigEnable
+	}
+
+	session, err := session.NewSessionWithOptions(session_config)
+	if err != nil{
+		panic(err)
```

Comment:
I think it would be better just return the error to the main function, gracefully terminate the process instead of just panic.